### PR TITLE
drivers: sensor: adxl345: Support for RTIO I2C stream

### PR DESCRIPTION
**Pull request by dimitrije-lilic**
Updated ADXL345 driver with RTIO I2C stream functionality.

Signed-off-by: Dimitrije Lilic <dimitrije.lilic@orioninc.com>

### CODE REVIEW

**Review by teburd — CHANGES_REQUESTED** [review]
It's actually quite nice that this mostly *just works* with spi or i2c, with some small flag changes. The pool size of sqe/cqe's is way too big though. Each submission queue entry is only limited to a parts cache line size, commonly 64 bytes! this is 64*64 bytes (4kb!) of submissions which is too large really when the largest transaction here is likely only 2-4 submissions long (128-256 bytes at most).

File: drivers/sensor/adi/adxl345/adxl345.c [hunk 1]
```diff
@@ -519,11 +519,17 @@ static int adxl345_init(const struct device *dev)
...
 #define ADXL345_RTIO_DEFINE(inst)                                    \
 	SPI_DT_IODEV_DEFINE(adxl345_iodev_##inst, DT_DRV_INST(inst),     \
 						SPI_WORD_SET(8) | SPI_TRANSFER_MSB |    \
 						SPI_MODE_CPOL | SPI_MODE_CPHA, 0U);     \
 	RTIO_DEFINE(adxl345_rtio_ctx_##inst, 64, 64);
+#elif CONFIG_I2C_RTIO
+#define ADXL345_RTIO_DEFINE(inst)                                    \
+	I2C_DT_IODEV_DEFINE(adxl345_iodev_##inst, DT_DRV_INST(inst));	\
+	RTIO_DEFINE(adxl345_rtio_ctx_##inst, 64, 64);
+#endif
```
Comment (MaureenHelm) [inline drivers/sensor/adi/adxl345/adxl345.c:532]: This won't work if there is one instance of the sensor on an I2C bus and another instance on a SPI bus

File: drivers/sensor/adi/adxl345/adxl345_stream.c [hunk 2]
```diff
@@ -216,6 +216,9 @@ static void adxl345_process_fifo_samples_cb(struct rtio *r, const struct rtio_sq
 							read_buf + data->fifo_total_bytes,
 							SAMPLE_SIZE, current_sqe);
 		data->fifo_total_bytes += SAMPLE_SIZE;
+#if defined(CONFIG_I2C_RTIO)
+		read_fifo_data->iodev_flags |= RTIO_IODEV_I2C_STOP | RTIO_IODEV_I2C_RESTART;
+#endif
```
Comment (MaureenHelm) [inline drivers/sensor/adi/adxl345/adxl345_stream.c:221]: I think you need to check if this instance of the sensor is on an I2C bus

File: drivers/sensor/adi/adxl345/adxl345.c [hunk 3]
```diff
@@ -519,11 +519,17 @@ static int adxl345_init(const struct device *dev)
...
+#ifdef CONFIG_SPI_RTIO
 #define ADXL345_RTIO_DEFINE(inst)                                    \
 	SPI_DT_IODEV_DEFINE(adxl345_iodev_##inst, DT_DRV_INST(inst),     \
 						SPI_WORD_SET(8) | SPI_TRANSFER_MSB |    \
 						SPI_MODE_CPOL | SPI_MODE_CPHA, 0U);     \
 	RTIO_DEFINE(adxl345_rtio_ctx_##inst, 64, 64);
+#elif CONFIG_I2C_RTIO
+#define ADXL345_RTIO_DEFINE(inst)                                    \
+	I2C_DT_IODEV_DEFINE(adxl345_iodev_##inst, DT_DRV_INST(inst));	\
+	RTIO_DEFINE(adxl345_rtio_ctx_##inst, 64, 64);
```
[the remark this answers is not in the document]
Comment (teburd) [reply drivers/sensor/adi/adxl345/adxl345.c:531]: Argh, yeah thats a bit of a thing isn't it, in that case its up to you, its a trade off of having to refill the submissions with a callback or having more submissions in the pool at a cost of memory.

**Review by teburd — DISMISSED** [review]

File: drivers/sensor/adi/adxl345/adxl345_stream.c [hunk 4]
```diff
@@ -342,6 +345,9 @@ static void adxl345_process_status1_cb(struct rtio *r, const struct rtio_sqe *sq
 	rtio_sqe_prep_read(read_fifo_data, data->iodev, RTIO_PRIO_NORM, data->fifo_ent, 1,
 						current_sqe);
 	read_fifo_data->flags = RTIO_SQE_CHAINED;
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
```
[the remark this answers is not in the document]
Comment (dimitrije-lilic) [reply drivers/sensor/adi/adxl345/adxl345_stream.c]: changed

File: drivers/sensor/adi/adxl345/adxl345_stream.c [hunk 5]
```diff
@@ -366,6 +372,9 @@ void adxl345_stream_irq_handler(const struct device *dev)
 	rtio_sqe_prep_read(read_status_reg, data->iodev, RTIO_PRIO_NORM, &data->status1, 1, NULL);
 	read_status_reg->flags = RTIO_SQE_CHAINED;
 
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
```
[the remark this answers is not in the document]
Comment (dimitrije-lilic) [reply drivers/sensor/adi/adxl345/adxl345_stream.c]: changed

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**